core/src/main/java/io/jenkins/pluginhealth/scoring/probes/ThirdPartyRepositoryDetectionProbe.java

```diff
@@ -0,0 +1,111 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Repository;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(value = ThirdPartyRepositoryDetectionProbe.ORDER)
+public class ThirdPartyRepositoryDetectionProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ThirdPartyRepositoryDetectionProbe.class);
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 100;
+    public static final String KEY = "third-party-repository-detection-probe";
+    final String hostName = "https://repo.jenkins-ci.org";
+//    final String parentPom = "https://raw.githubusercontent.com/jenkinsci/plugin-pom/master/pom.xml";
+    final String parentPom = "https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+        Set<Repository> allRepositories = new HashSet<>();
+
+        try {
+            InputStream inputStream = new FileInputStream(context.getScmRepository() + "/pom.xml");
+            Reader reader = new InputStreamReader(inputStream, "UTF-8");
```

Comment:
Please, look at `try-with-resources`, what it is, why and how you can use it.

```suggestion
        try(InputStream inputStream = new FileInputStream(context.getScmRepository() + "/pom.xml");
              Reader reader = new InputStreamReader(inputStream, "UTF-8")) {
```